File: ruby/word-count/word_count.rb
class Phrase
  attr_reader :words

  def initialize(phrase)
    @words = phrase.downcase.scan /\b[\w\']+\b/
  end

  def word_count
    words.uniq.each_with_object({}) do |word, word_count|
      word_count[word] = words.count(word)
    end
  end
end

File: elixir/word-count/lib/word_count.ex
defmodule WordCount do
  @doc """
  Count the number of words in the sentence.

  Words are compared case-insensitively.
  """
  @spec count(String.t()) :: map
  def count(sentence) do
  	words = Regex.scan(~r/[[:alnum:]|\-|ö]+/,	String.downcase(sentence))
		Enum.into(words, %{}, fn word ->
			{Enum.at(word, 0), Enum.count(words, fn w -> w == word end)}
		end)
  end
end
